fmt/src/header/parser.rs

```diff
@@ -71,29 +73,33 @@ pub fn parse_header(
     };
 
     // 2. has header
-    let existing_header = {
+    let mut existing_header = || -> bool {
```

Comment:
Why this change needed? I tend to avoid "top" level mut.

This block is somewhat a function but I rush it without handling argument passing.

Reply:
I added this for the sake of keeping to code clean. Otherwise I needed to add a new `if` block which will push the indentation further 💀 Now I can early return from this closure.

But I can remove it and simply use and `if` if you prefer that.

Reply:
I factor out this block into a function then.